Shop/ShopItem.gd

```diff
@@ -0,0 +1,23 @@
+extends Control
+class_name ShopItem
+
+## base class for all items sold in the shop
+## The classes deriving from this are named shopitem[item which is sold]
+
+@export var cost : int
+@export var priceLabel : Label
+
+func _ready() -> void:
+	priceLabel.text = str(cost)
+
+func on_shop_item_clicked() ->void:
+	if(InventoryManager.gold_component.can_afford(cost)):
+		queue_free()
+		InventoryManager.gold_component.lose_gold(cost)
+		_give_player_item()
+	else:
+		print("can't afford")
```

Comment:
I'd say send something to the scene here to inform the user that they don't have enough money to purchase the item. Probably a debug variable in DEBUG_VAR to let us purchase w/o having to have gold, for testing purposes!

Reply:
I'll make a debug function which gives a lot of gold instead, as that is easier and should do basicly the same thing